specifyweb/frontend/js_src/lib/components/Statistics/hooks.tsx

```diff
@@ -76,6 +97,44 @@ function backEndStatPromiseGenerator(
   );
 }
 
+function dynamicEphermeralPromiseGenerator(
+  dynamicEphemeralFieldSpecs: RA<DynamicQuerySpec>
+): IR<() => Promise<RA<string> | undefined>> {
+  return Object.fromEntries(
+    dynamicEphemeralFieldSpecs.map(({ key, spec }) => [
+      key,
+      async () =>
+        throttledPromise<RA<string> | undefined>(
+          'queryStats',
+          async () =>
```

Comment:
Could you add this comment in the code here?

```ts
// REFACTOR: use runQuery() function once merged with xml-editor
```

Reply:
added!